### ISSUE TITLE
The output wav file of the speech enhancement task is distorted.

### ISSUE THREAD

**Issue by takashin3391** [espnet/espnet#4245]
Thank you for all your help.
I tried the vctk_noisy voice enhancement task in egs2.
However, the output wav file (e.g. 
 `espnet/egs2/vctk_noisy/enh1/exp/enh_train_raw/enhanced_tt_2spk/logdir/output.1/wavs/1/p232_001.wav`) is very distorted.

<img width="1237" alt="p232_001" src="https://user-images.githubusercontent.com/3216508/162367827-d2ed848b-57bd-4e6f-8c7d-79d7f2f68eb9.PNG">

The loss is decreasing, so it seems to be learning.

Is there something wrong with my settings?
I would like to know why the audio is distorted and what to do about it.

**Comment by takashin3391** [comment]
run.sh and enh.sh are attached.
(extension changed from .sh to .txt)

[run.txt](https://github.com/espnet/espnet/files/8449401/run.txt)
[enh.txt](https://github.com/espnet/espnet/files/8449402/enh.txt)

**Comment by Emrys365** [comment]
Could you trying enhancing some samples from training and validation sets? I wonder if they also sound distorted.
If so, there might be some mismatch/bug in the inference stage.
Otherwise, it might be due to the overfitting issue.

**Comment by Emrys365** [comment]
> I then replaced the vctk_noisy data with the own data.

Do you mean replacing only test data or both training and test data?

**Comment by takashin3391** [comment]
> Do you mean replacing only test data or both training and test data?

I mean replacing both training and test data.

**Comment by Emrys365** [comment]
> I mean replacing both training and test data.

Okay, then what is the amount of training data (say ____ hours)?

**Comment by takashin3391** [comment]
> what is the amount of training data (say ____ hours)?
The amount of training data is about 8.5 hours.

**Comment by Emrys365** [comment]
> > what is the amount of training data (say ____ hours)?
> 
> The amount of training data is about 8.5 hours.

I think it is too small. If possible, please increase the amount to at least tens of hours.

**Comment by takashin3391** [comment]
@Emrys365 
Hello.
I have increased my training data to 30 hours and trained again.
(30 hours is almost the same amount of data as vctk_noisy)
However, the output audio was distorted.

I ran vctk_noisy again, but the output audio was distorted.
(Previously the output audio was not distorted)
It seems that once the output audio is distorted, the output results are distorted afterwards.

Importantly, when I train my data in another environment, the output audio is not distorted.

Why does this happen?
Is there a way to learn with cleanliness?

**Comment by Emrys365** [comment]
> @Emrys365 
> Hello.
> I have increased my training data to 30 hours and trained again.
> (30 hours is almost the same amount of data as vctk_noisy)
> However, the output audio was distorted.
> 
> I ran vctk_noisy again, but the output audio was distorted.
> (Previously the output audio was not distorted)
> It seems that once the output audio is distorted, the output results are distorted afterwards.
> 
> Importantly, when I train my data in another environment, the output audio is not distorted.
> 
> Why does this happen?
> Is there a way to learn with cleanliness?

Did you somehow mix the vctk_noisy data and your own data? It could cause inconsistency in different runs of the same experiment.
> It is better to use separate recipe directories for different data, if you didn't.

Could you post your learning curves of each experiment?

**Comment by takashin3391** [comment]
@Emrys365 
Thank you very much for your reply.

I have figured out why the output audio is distorted.
If I change `inference_args="--normalize_output_wav true"` to `inference_args="--normalize_output_wav false"` in `enh.sh`, the output audio is distorted.

What does normalization mean?

**Comment by Emrys365** [comment]
> @Emrys365 
> Thank you very much for your reply.
> 
> I have figured out why the output audio is distorted.
> If I change `inference_args="--normalize_output_wav true"` to `inference_args="--normalize_output_wav false"` in `enh.sh`, the output audio is distorted.
> 
> What does normalization mean?

I see. You could check out `espnet2/bin/enh_inference.py` to understand how it works. Basically, it rescales the estimated audio to to in range [-1.0, 1.0]. If your model is trained with some scale-invariant loss, it is likely the model output will have a large magnitude, and normalization is needed in this case.
